dpnp/dpnp_array.py

```diff
@@ -292,8 +292,10 @@ def __str__(self):
 
         """
 
-        return str(self.asnumpy())
+        return self._array_obj.__str__()
 
+    def __repr__(self):
```

Comment:
It would be better to move the method at line 250, where we have an appropriate comment (to keep the lexicographical order)
```
# '__repr__',
```